### ISSUE TITLE
Bezier commands

### ISSUE THREAD

**Issue by Le2222** [drawpile/Drawpile#467]
I have a question, maybe an issue:

In Krita/Inkscape, using the bezier tool, when I do a curve segment with 'Click+Drag' it rasterizes the previous segment and prepares the next one (a virtual/movable preview line of the next segment appears). It does happen in the same way in Drawpile (fine).

The problem is: In Krita/Inkscape, if I'm done and I don't want to draw (rasterize) the very last virtual segment I just press 'Enter' (Inkscape also accepts 'Right Click' to this function: it have the 'Right Click' avaible because just like Photoshop it uses 'Backspace' and 'Ctrl+Z' instead of 'Right Click' to remove the last anchor). But in Drawpile, I don't know how to do it (finish the line ignoring the lastest virtual segment). Is there a way to do it? I tried to press 'Enter' and 'Esc' and mouse buttons and etc. but nothing happens.

Example, press 'Enter' (or maybe 'Right Click', if 'delete previous anchor' function pass to the 'Backspace'/'Ctrl+Z' keys) would render the segment 1 (raster, blue) and ignore the segment 2 (virtual, red):

![example](https://i.gyazo.com/a63513fad20024c621aace1e080d82e1.png)

Example's expected result:

![example result](https://i.gyazo.com/229b5518ce8479c4004dce707e3fefb1.png)

**Comment by callaa** [comment]
Currently the only way to finish a line is to draw a segment with no curve. In some situations it does result in a visible extra segment, so a keyboard command to finish the stroke would indeed be useful.

**Comment by Le2222** [comment]
I also think the fact of ctrl+z undo the previous move before the bezier (ignoring the bezier steps) is a confusing behavior.

Example: If i draw a square then a circle then im doing bezier curves and press ctrl+z, it undoes the circle instead of the last curve.

**Comment by callaa** [comment]
These changes are now implemented. Hitting `Return` now commits the drawing the same way it does Krita and in your example. Ctrl+Z now undos the latest curve segment.

I also took the opportunity to refactor the selection tools to use these new multipart drawing interfaces I added for the bezier tool. Ctrl+Z now first resets a moved selection to its original state, then cancels it. The `Return` key also now works with selections.
